packages/syft/src/syft/service/queue/zmq_queue.py

```diff
@@ -570,6 +584,17 @@ def process_worker(self, address: bytes, msg: list[bytes]) -> None:
 
     def delete_worker(self, worker: Worker, disconnect: bool) -> None:
         """Deletes worker from all data structures, and deletes worker."""
+        # relative
+        from ...service.worker.worker_service import WorkerService
+
+        worker_service = cast(
+            WorkerService, self.auth_context.node.get_service(WorkerService)
+        )
+        worker_service._delete(
```

Comment:
may need to check if job_id is present then skip deleting. When no job_id is present, it means that worker is idle and we can safely delete it.